[2024-02-27T15:14:03.620+0700] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: missed_deadline_job_dag.spark_ingestion_to_s3 manual__2024-02-27T07:38:24.396245+00:00 [queued]>
[2024-02-27T15:14:03.626+0700] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: missed_deadline_job_dag.spark_ingestion_to_s3 manual__2024-02-27T07:38:24.396245+00:00 [queued]>
[2024-02-27T15:14:03.626+0700] {taskinstance.py:2170} INFO - Starting attempt 2 of 2
[2024-02-27T15:14:03.637+0700] {taskinstance.py:2191} INFO - Executing <Task(BashOperator): spark_ingestion_to_s3> on 2024-02-27 07:38:24.396245+00:00
[2024-02-27T15:14:03.639+0700] {standard_task_runner.py:60} INFO - Started process 95389 to run task
[2024-02-27T15:14:03.641+0700] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'missed_deadline_job_dag', 'spark_ingestion_to_s3', 'manual__2024-02-27T07:38:24.396245+00:00', '--job-id', '173', '--raw', '--subdir', 'DAGS_FOLDER/missed_deadlinee_dag.py', '--cfg-path', '/tmp/tmppq0oxm3s']
[2024-02-27T15:14:03.643+0700] {standard_task_runner.py:88} INFO - Job 173: Subtask spark_ingestion_to_s3
[2024-02-27T15:14:03.697+0700] {task_command.py:423} INFO - Running <TaskInstance: missed_deadline_job_dag.spark_ingestion_to_s3 manual__2024-02-27T07:38:24.396245+00:00 [running]> on host MinhNhut.
[2024-02-27T15:14:03.784+0700] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='missed_deadline_job_dag' AIRFLOW_CTX_TASK_ID='spark_ingestion_to_s3' AIRFLOW_CTX_EXECUTION_DATE='2024-02-27T07:38:24.396245+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-02-27T07:38:24.396245+00:00'
[2024-02-27T15:14:03.785+0700] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2024-02-27T15:14:03.785+0700] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'python /home/nhut/etl-***/pipeline/spark_ingestion_to_s3.py']
[2024-02-27T15:14:03.790+0700] {subprocess.py:86} INFO - Output:
[2024-02-27T15:14:05.283+0700] {subprocess.py:93} INFO - 24/02/27 15:14:05 WARN Utils: Your hostname, MinhNhut resolves to a loopback address: 127.0.1.1; using 172.28.89.67 instead (on interface eth0)
[2024-02-27T15:14:05.286+0700] {subprocess.py:93} INFO - 24/02/27 15:14:05 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
[2024-02-27T15:14:15.459+0700] {subprocess.py:93} INFO - :: loading settings :: url = jar:file:/home/nhut/etl-***/.venv/lib/python3.10/site-packages/pyspark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2024-02-27T15:14:15.543+0700] {subprocess.py:93} INFO - Ivy Default Cache set to: /home/nhut/.ivy2/cache
[2024-02-27T15:14:15.544+0700] {subprocess.py:93} INFO - The jars for the packages stored in: /home/nhut/.ivy2/jars
[2024-02-27T15:14:15.548+0700] {subprocess.py:93} INFO - org.apache.hadoop#hadoop-aws added as a dependency
[2024-02-27T15:14:15.549+0700] {subprocess.py:93} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-7a1878f9-a991-405d-b4c5-e9f4fa9dc8b8;1.0
[2024-02-27T15:14:15.549+0700] {subprocess.py:93} INFO - 	confs: [default]
[2024-02-27T15:14:15.660+0700] {subprocess.py:93} INFO - 	found org.apache.hadoop#hadoop-aws;3.3.2 in spark-list
[2024-02-27T15:14:15.680+0700] {subprocess.py:93} INFO - 	found com.amazonaws#aws-java-sdk-bundle;1.11.1026 in spark-list
[2024-02-27T15:14:15.698+0700] {subprocess.py:93} INFO - 	found org.wildfly.openssl#wildfly-openssl;1.0.7.Final in central
[2024-02-27T15:14:15.715+0700] {subprocess.py:93} INFO - :: resolution report :: resolve 158ms :: artifacts dl 8ms
[2024-02-27T15:14:15.715+0700] {subprocess.py:93} INFO - 	:: modules in use:
[2024-02-27T15:14:15.715+0700] {subprocess.py:93} INFO - 	com.amazonaws#aws-java-sdk-bundle;1.11.1026 from spark-list in [default]
[2024-02-27T15:14:15.715+0700] {subprocess.py:93} INFO - 	org.apache.hadoop#hadoop-aws;3.3.2 from spark-list in [default]
[2024-02-27T15:14:15.716+0700] {subprocess.py:93} INFO - 	org.wildfly.openssl#wildfly-openssl;1.0.7.Final from central in [default]
[2024-02-27T15:14:15.716+0700] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2024-02-27T15:14:15.716+0700] {subprocess.py:93} INFO - 	|                  |            modules            ||   artifacts   |
[2024-02-27T15:14:15.716+0700] {subprocess.py:93} INFO - 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2024-02-27T15:14:15.716+0700] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2024-02-27T15:14:15.716+0700] {subprocess.py:93} INFO - 	|      default     |   3   |   0   |   0   |   0   ||   3   |   0   |
[2024-02-27T15:14:15.716+0700] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2024-02-27T15:14:15.718+0700] {subprocess.py:93} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-7a1878f9-a991-405d-b4c5-e9f4fa9dc8b8
[2024-02-27T15:14:15.719+0700] {subprocess.py:93} INFO - 	confs: [default]
[2024-02-27T15:14:15.723+0700] {subprocess.py:93} INFO - 	0 artifacts copied, 3 already retrieved (0kB/5ms)
[2024-02-27T15:14:15.885+0700] {subprocess.py:93} INFO - 24/02/27 15:14:15 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-02-27T15:14:16.038+0700] {subprocess.py:93} INFO - Setting default log level to "WARN".
[2024-02-27T15:14:16.038+0700] {subprocess.py:93} INFO - To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
[2024-02-27T15:14:16.881+0700] {subprocess.py:93} INFO - 24/02/27 15:14:16 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2024-02-27T15:14:23.614+0700] {subprocess.py:93} INFO - 24/02/27 15:14:23 WARN MetricsConfig: Cannot locate configuration: tried hadoop-metrics2-s3a-file-system.properties,hadoop-metrics2.properties
[2024-02-27T15:14:49.241+0700] {local_task_job_runner.py:302} WARNING - State of this instance has been externally set to restarting. Terminating instance.
[2024-02-27T15:14:49.244+0700] {process_utils.py:131} INFO - Sending 15 to group 95389. PIDs of all processes in the group: [95397, 95398, 95389]
[2024-02-27T15:14:49.244+0700] {process_utils.py:86} INFO - Sending the signal 15 to group 95389
[2024-02-27T15:14:49.244+0700] {taskinstance.py:2450} ERROR - Received SIGTERM. Terminating subprocesses.
[2024-02-27T15:14:49.244+0700] {subprocess.py:104} INFO - Sending SIGTERM signal to process group
[2024-02-27T15:14:49.261+0700] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/nhut/etl-airflow/.venv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/nhut/etl-airflow/.venv/lib/python3.10/site-packages/airflow/operators/bash.py", line 203, in execute
    result = self.subprocess_hook.run_command(
  File "/home/nhut/etl-airflow/.venv/lib/python3.10/site-packages/airflow/hooks/subprocess.py", line 91, in run_command
    for raw_line in iter(self.sub_process.stdout.readline, b""):
  File "/home/nhut/etl-airflow/.venv/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 2452, in signal_handler
    raise AirflowException("Task received SIGTERM signal")
airflow.exceptions.AirflowException: Task received SIGTERM signal
[2024-02-27T15:14:49.264+0700] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=missed_deadline_job_dag, task_id=spark_ingestion_to_s3, execution_date=20240227T073824, start_date=20240227T081403, end_date=20240227T081449
[2024-02-27T15:14:49.277+0700] {standard_task_runner.py:107} ERROR - Failed to execute job 173 for task spark_ingestion_to_s3 (Task received SIGTERM signal; 95389)
[2024-02-27T15:14:49.296+0700] {process_utils.py:79} INFO - Process psutil.Process(pid=95397, status='terminated', started='15:14:03') (95397) terminated with exit code None
[2024-02-27T15:14:49.297+0700] {process_utils.py:79} INFO - Process psutil.Process(pid=95389, status='terminated', exitcode=1, started='15:14:03') (95389) terminated with exit code 1
[2024-02-27T15:14:49.723+0700] {process_utils.py:79} INFO - Process psutil.Process(pid=95398, status='terminated', started='15:14:03') (95398) terminated with exit code None
